src/zabapgit_gui_router.prog.abap

```diff
@@ -85,11 +86,17 @@ CLASS lcl_gui_router IMPLEMENTATION.
         ei_page  = get_page_background( lv_key ).
         ev_state = gc_event_state-new_page.
       WHEN gc_action-go_diff.                         " Go Diff page
-        ei_page  = get_page_diff( iv_getdata ).
-        ev_state = gc_event_state-new_page.
-      WHEN gc_action-go_stage.                        " Go Staging page
-        ei_page  = get_page_stage( lv_key ).
+        ei_page  = get_page_diff(
+          iv_getdata   = iv_getdata
+          iv_prev_page = iv_prev_page ).
         ev_state = gc_event_state-new_page_w_bookmark.
+      WHEN gc_action-go_stage.                        " Go Staging page
+        ei_page  = get_page_stage( iv_getdata ).
+        IF iv_prev_page = 'PAGE_DIFF'.
```

Comment:
I think we need to look into if this special case can be avoided, but we can do that later, after this PR is merged